### ISSUE TITLE
Headless geofence fired, not the shape expected

### ISSUE THREAD

**Issue by JAStanton** [transistorsoft/react-native-background-geolocation#592]
Hello, I recently upgraded to plugin version `2.13.4`, before this I was on version `2.10.1`. I have headless mode enabled on when I'm receiving a geofence fired event it's not the shape that I expected.

I thought geofence metadata was nested under `params.geofence`, but it appears that metadata is directly on `params`, is that a recent change? 

This is a log from the device that fired the event
```json
[BackgroundGeolocation HeadlessService]  {
    "params": {
        "action": "ENTER",
        "identifier": "REDACTED~~60ddc503-582c-415c-9aa9-97879ddb50a7~~aeb969e3-66f8-4b91-a964-c7b0fd2027d5",
        "extras": {
            "redactedId": "60ddc503-582c-415c-9aa9-97879ddb50a7",
            "redacted2Id": "aeb969e3-66f8-4b91-a964-c7b0fd2027d5",
            "createdAt": "2018-10-22T10:40:49.293Z",
            "notifyOnEntry": true,
            "notifyOnExit": true
        },

        "location": {
            "extras": {},
            "geofence": {
                "extras": {
                    "redactedId": "60ddc503-582c-415c-9aa9-97879ddb50a7",
                    "redacted2Id": "aeb969e3-66f8-4b91-a964-c7b0fd2027d5",
                    "createdAt": "2018-10-22T10:40:49.293Z",
                    "notifyOnEntry": true,
                    "notifyOnExit": true
                },
                "action": "ENTER",
                "identifier": "REDACTED~~60ddc503-582c-415c-9aa9-97879ddb50a7~~aeb969e3-66f8-4b91-a964-c7b0fd2027d5"
            },
            "battery": {
                "level": 0.74,
                "is_charging": false
            },
            "coords": {
                "heading": -1,
                "speed": 0,
                "accuracy": 34.9,
                "longitude": "redacted",
                "altitude": -40,
                "latitude": "redacted7"
            },
            "odometer": "118067.0",
            "timestamp": "2018-10-22T14:34:55.213Z",
            "uuid": "ca59a498-c425-41d2-9480-4e96d595de95",
            "activity": {
                "confidence": 100,
                "type": "still"
            },
            "is_moving": false,
            "event": "geofence"
        }
    },
    "name": "geofence"
}
```

## Your Environment
* Plugin version: 2.13.4
* Platform: Android
* OS version: 7.0
* Device manufacturer / model: LG-LS777 
* React Native version (`react-native -v`): 0.53
* Plugin config
```
 { config:
    { disableMotionActivityUpdates: true,
      preventSuspend: true,
      startOnBoot: true,
      stopOnTerminate: false,
      foregroundService: true,
      enableHeadless: true,
      allowIdenticalLocations: true,
      disableElasticity: false,
      distanceFilter: 0,
      stationaryRadius: 25,
      stopTimeout: 5,
      notificationTitle: 'redacted',
      notificationText: 'redacted',
      notificationSmallIcon: 'drawable/ic_stat_name',
      notificationColor: '#F65335',
      notificationChannelName: 'redacted',
      notificationPriority: -2,
      logLevel: 5,
      desiredAccuracy: -1,
      useSignificantChangesOnly: false,
      fastestLocationUpdateInterval: 60000,
      locationUpdateInterval: 60000,
      heartbeatInterval: 60,
    }
}
```

## Expected Behavior
1. Fire geofence in headless mode
2. print out event.params.geofence (should be geofence data like: action, identifier, extras...)

## Actual Behavior
1. Fire geofence in headless mode
2. print out event.params.geofence (will be undefined)

## Steps to Reproduce

1. Fire geofence in headless mode
2. print out event.params.geofence (will be undefined)

## Context
I upgraded RNBGA version, trying to figure out where geofence metadata should be

## Debug logs
Added 1 log above

**Comment by christocracy** [comment]
Yes, I did make a big change to Headless, where I removed the `JobService` mechanism of firing headless-events to use a more direct mechanism.  I don't recall exactly what I did to the Geofence event, but it should be standardized now between headless and normal state.

The current schema should match the docs (which have been refactored).  See [GeofenceEvent](https://transistorsoft.github.io/react-native-background-geolocation/interfaces/_react_native_background_geolocation_.geofenceevent.html)
